bin/run-circleci-tests

```diff
@@ -4,6 +4,18 @@ set -e
 
 make install_system_dependencies
 
+# Ensure Redis is up and running before starting to run any tests:
+# In the past, individual tests from our unit test suite
+# would sometimes fail with a "ConnectionRefusedError"
+# caused by Redis not starting for the container running the tests.
+# The solution that we use below comes from the following support thread:
+# https://discuss.circleci.com/t/redis-server-problems-unable-to-connect-localhost-6379/4040
+until bash -c "sudo service redis-server status"; do
```

Comment:
@jbzdak Where can I see the maximum build time for our tests? In any case, I'm going to limit the time that we wait for Redis to start to 5 minutes. That still leaves plenty of time to start it in the rare case where it fails to start automatically for a container. Also, if the state of a container makes it impossible to start Redis within 5 minutes, it's probably not worth to keep trying :)